File: ws-lib/src/main/java/com/tim/iot/ws/entity/WebSocketInfo.java
package com.tim.iot.ws.entity;

import java.io.Serializable;
import okhttp3.WebSocket;
import okio.ByteString;

/**
 * @author Tell.Tim
 * @date 2019/12/2 15:36
 */
public class WebSocketInfo implements Serializable {
    private static final long serialVersionUID = 1714523747523892210L;
    private WebSocket mWebSocket;
    private String simpleMsg;
    private ByteString byteStringMsg;
    private boolean connected;
    private boolean reconnect;

    public WebSocket getWebSocket() {
        return mWebSocket;
    }

    public void setWebSocket(WebSocket mWebSocket) {
        this.mWebSocket = mWebSocket;
    }

    public boolean isConnected() {
        return connected;
    }

    public void setConnected(boolean connected) {
        this.connected = connected;
    }

    public boolean isReconnect() {
        return reconnect;
    }

    public void setReconnect(boolean reconnect) {
        this.reconnect = reconnect;
    }

    public String getSimpleMsg() {
        return simpleMsg;
    }

    public void setSimpleMsg(String simpleMsg) {
        this.simpleMsg = simpleMsg;
    }

    public ByteString getByteStringMsg() {
        return byteStringMsg;
    }

    public void setByteStringMsg(ByteString byteStringMsg) {
        this.byteStringMsg = byteStringMsg;
    }

    public WebSocketInfo reset() {
        this.mWebSocket = null;
        this.simpleMsg = null;
        this.byteStringMsg = null;
        this.connected = false;
        this.reconnect = false;
        return this;
    }

    public static WebSocketInfo createReconnect() {
        WebSocketInfo socketInfo = new WebSocketInfo();
        socketInfo.reconnect = true;
        return socketInfo;
    }

    public static WebSocketInfo createConnected(WebSocket webSocket) {
        WebSocketInfo socketInfo = new WebSocketInfo();
        socketInfo.connected = true;
        socketInfo.mWebSocket = webSocket;
        return socketInfo;
    }

    public static WebSocketInfo createMsg(WebSocket webSocket, String simpleMsg) {
        WebSocketInfo socketInfo = new WebSocketInfo();
        socketInfo.simpleMsg = simpleMsg;
        socketInfo.mWebSocket = webSocket;
        return socketInfo;
    }

    public static WebSocketInfo createByteStringMsg(WebSocket webSocket, ByteString byteMsg) {
        WebSocketInfo socketInfo = new WebSocketInfo();
        socketInfo.byteStringMsg = byteMsg;
        socketInfo.mWebSocket = webSocket;
        return socketInfo;
    }
}

File: app/src/main/java/com/tim/iot/auth/entity/WebSocketInfo.kt
package com.tim.iot.auth.entity

import java.io.Serializable
import okhttp3.WebSocket
import okio.ByteString

/**
 * @author Tell.Tim
 * @date 2019/12/2 15:36
 */
class WebSocketInfo : Serializable {
    var webSocket: WebSocket? = null
    var simpleMsg: String? = null
    var byteStringMsg: ByteString? = null
    var isConnected: Boolean = false
    var isReconnect: Boolean = false
    var isPreConnect: Boolean = false

    fun reset(): WebSocketInfo {
        this.webSocket = null
        this.simpleMsg = null
        this.byteStringMsg = null
        this.isConnected = false
        this.isReconnect = false
        return this
    }

    companion object {
        private const val serialVersionUID = 1714523747523892210L

        fun createReconnect(): WebSocketInfo {
            val socketInfo = WebSocketInfo()
            socketInfo.isReconnect = true
            return socketInfo
        }

        fun createConnected(webSocket: WebSocket): WebSocketInfo {
            val socketInfo = WebSocketInfo()
            socketInfo.isConnected = true
            socketInfo.webSocket = webSocket
            return socketInfo
        }

        fun createPreConnect(webSocket: WebSocket): WebSocketInfo {
            val socketInfo = WebSocketInfo()
            socketInfo.isPreConnect = true
            socketInfo.webSocket = webSocket
            return socketInfo
        }

        fun createMsg(webSocket: WebSocket, simpleMsg: String): WebSocketInfo {
            val socketInfo = WebSocketInfo()
            socketInfo.simpleMsg = simpleMsg
            socketInfo.webSocket = webSocket
            return socketInfo
        }

        fun createByteStringMsg(webSocket: WebSocket, byteMsg: ByteString): WebSocketInfo {
            val socketInfo = WebSocketInfo()
            socketInfo.byteStringMsg = byteMsg
            socketInfo.webSocket = webSocket
            return socketInfo
        }
    }
}
